crates/oxc_linter/src/rules/eslint/no_regex_spaces.rs

```diff
@@ -0,0 +1,253 @@
+    /// then specify how many spaces are expected using a quantifier.
+    ///
+    /// ```javascript
+    /// var re = /foo {3}bar/;
+    /// ```
+    ///
+    /// ### Example
+    /// ```javascript
+    /// var re = /foo   bar/;
+    /// ```
+    NoRegexSpaces,
+    restriction,
+);
+
+impl Rule for NoRegexSpaces {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        match node.kind() {
+            AstKind::RegExpLiteral(lit) => {
+                if let Some(span) = Self::find_literal_to_report(lit) {
+                    ctx.diagnostic(NoRegexSpacesDiagnostic(span)); // /a  b/
+                }
+            }
+
+            AstKind::CallExpression(expr) if Self::is_regexp_call_expression(expr) => {
+                if let Some(span) = Self::find_expr_to_report(&expr.arguments) {
+                    ctx.diagnostic(NoRegexSpacesDiagnostic(span)); // RegExp('a  b')
+                }
+            }
+
+            AstKind::NewExpression(expr) if Self::is_regexp_new_expression(expr) => {
+                if let Some(span) = Self::find_expr_to_report(&expr.arguments) {
+                    ctx.diagnostic(NoRegexSpacesDiagnostic(span)); // new RegExp('a  b')
+                }
+            }
+
+            _ => {}
+        }
+    }
+}
+
+impl NoRegexSpaces {
+    fn find_literal_to_report(lit: &RegExpLiteral) -> Option<Span> {
+        if Self::has_exempted_char_class(&lit.regex.pattern) {
+            return None;
+        }
+
+        if Self::has_target_consecutive_spaces(&lit.regex.pattern) {
+            return Some(lit.span);
+        }
+
+        None
+    }
+
+    fn find_expr_to_report(args: &Vec<'_, Argument<'_>>) -> Option<Span> {
+        if let Some(Argument::Expression(expr)) = args.get(1) {
+            if !expr.is_string_literal() {
+                return None; // skip on indeterminate flag, e.g. RegExp('a  b', flags)
+            }
+        }
+
+        if let Some(Argument::Expression(Expression::StringLiteral(pattern))) = args.get(0) {
+            if Self::has_exempted_char_class(&pattern.value) {
+                return None; // skip spaces inside char class, e.g. RegExp('[  ]')
+            }
+
+            if !Self::has_unescaped_consecutive_spaces(&pattern.value) {
+                return None; // skip if not literally consecutive, e.g. RegExp(' \ ')
+            }
+
+            if Self::has_target_consecutive_spaces(&pattern.value) {
+                return Some(pattern.span);
+            }
+        }
+
+        None
+    }
+
+    fn has_unescaped_consecutive_spaces(pattern: &Atom) -> bool {
+        regex::Regex::new(" {2}").unwrap().is_match(pattern)
+    }
+
+    fn has_target_consecutive_spaces(pattern: &Atom) -> bool {
+        // https://github.com/eslint/eslint/blob/485ec7d08ed2040c292f52bf9b9152f6c8ef4809/lib/rules/no-regex-spaces.js#L93
+        regex::Regex::new("( {2,})(?: [+*{?]|[^+*{?]|$)").unwrap().is_match(pattern)
```

Comment:
This regex is initialized every time it is called, which is really expensive. You need to put this into a `lazy_static`, take a look other places where regexes are constructed.

Reply:
What does this regex do actually? Are we able to rewrite it in simple loops?